### PR TITLE
Fix #415 by adding QUnit.extend method

### LINKED ISSUE #415: QUnit Interface: Implement QUnit.extend
This is a must need, I will soon put up a patch for this.

### PR DESCRIPTION
**Pull request by apsdehal**
Add QUnit.extend method for QUnit interface along with tests.

### CODE REVIEW

File: lib/interfaces/qunit.js [hunk 1]
```diff
@@ -151,6 +151,23 @@ define([
 			testTimeout: Infinity
 		},
 
+		extend: function (a, b, undefOnly) {
+			for (var prop in b) {
+				if (b.hasOwnProperty(prop)) {
+					// Avoid "Member not found" error in IE8 caused by messing with window.constructor
```
Comment (csnover) [inline lib/interfaces/qunit.js]: ie8 is not supported so this is not necessary

File: lib/interfaces/qunit.js [hunk 2]
```diff
@@ -151,6 +151,23 @@ define([
...
+					if (!(prop === "constructor" && a === window)) {
+						if (b[ prop ] === undefined) {
```
Comment (csnover) [inline lib/interfaces/qunit.js]: no spaces in array accessors please :)

File: tests/unit/lib/interfaces/qunit.js [hunk 3]
```diff
@@ -351,6 +351,18 @@ define([
 			}
 		},
 
+		'extend': {
+			'should have a working expect': function () {
```
Comment (csnover) [inline tests/unit/lib/interfaces/qunit.js]: this test doesn’t test the various `undefined` handling branches so is not complete.

File: tests/unit/lib/interfaces/qunit.js [hunk 4]
```diff
@@ -351,6 +351,18 @@ define([
...
+				QUnit.extend( QUnit.assert, {
```
Comment (csnover) [inline tests/unit/lib/interfaces/qunit.js]: this test is mutating objects that are not reset! (QUnit.assert)

Comment (csnover) [inline tests/unit/lib/interfaces/qunit.js]: use a new object scoped to the test to test this utility function, there is no reason to run it against anything constructed outside of the test function.

Comment (apsdehal) [inline tests/unit/lib/interfaces/qunit.js]: Fixed!

**Comment by csnover** [comment]
Thanks for the patch!

**Pull request closed without merging**